### PR TITLE
Add TransactionOptions

### PR DESCRIPTION
**Pull request by tom-andersen**
Add TransactionOptions parameter to runTransaction method. This will enable control over maximum attempts to commit before transaction fails.

### CODE REVIEW

File: packages/firestore/src/lite-api/transaction_options.ts [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/**
+ * Options to customize transaction behavior.
+ */
+export declare interface TransactionOptions {
+  /** maximum number of attempts to commit, after which transaction fails. Default is 5. */
```
Comment (wu-hui) [inline packages/firestore/src/lite-api/transaction_options.ts:22]: Capital `M`

File: packages/firestore/test/integration/api_internal/transaction.test.ts [hunk 2]
```diff
@@ -17,177 +17,190 @@
...
   runTransaction,
-  setDoc
+  setDoc,
+  TransactionOptions
 } from '../util/firebase_export';
 import { apiDescribe, withTestDb } from '../util/helpers';
 import { asyncQueue } from '../util/internal_helpers';
 
-apiDescribe(
+apiDescribe.only(
```
Comment (wu-hui) [inline packages/firestore/test/integration/api_internal/transaction.test.ts:34]: revert `.only`.

**Review by wu-hui — APPROVED** [review]
Nice PR!

File: packages/firestore/src/lite-api/transaction_options.ts (hunk 1 above)
Comment (tom-andersen) [reply packages/firestore/src/lite-api/transaction_options.ts:22]: Done

**Comment by wu-hui** [comment]
LGTM.

**Review by wu-hui — APPROVED** [review]

File: .changeset/lazy-nails-guess.md [hunk 3]
```diff
@@ -0,0 +1,5 @@
+---
+'@firebase/firestore': patch
+---
+
+Add TransactionOptions to runTransaction
```
Comment (hsubox76) [inline .changeset/lazy-nails-guess.md:5]: Can you add backticks to the 2 API names and add "param" after TransactionOptions?

Comment (tom-andersen) [reply .changeset/lazy-nails-guess.md]: Done

**Review by hsubox76 — APPROVED** [review]

**Pull request merged**